### PR TITLE
add: Firestore database

### PR DESCRIPTION
**Pull request by saurabhchirde**
Features added

1. Database integration on firestore.
2. New Achievements after 7 consecutive wins.

quiz-operation

### CODE REVIEW

**Comment by Vishnu-Aithal** [comment]
test review

**Review by Vishnu-Aithal — COMMENTED** [review]
Test comment

**Comment by thesudeshdas** [comment]
test review

**Review by thesudeshdas — COMMENTED** [review]
LGTM

**Review by yashodaAgarwal — APPROVED** [review]
UI looks great.Clean code, Can work on some below given suggestion. Rest Good to merge

File: src/Context/Network/NetworkContext.jsx [hunk 1]
```diff
@@ -20,6 +28,46 @@ const NetworkProvider = ({ children }) => {
...
+  // add user to database after log in or signup
+  // const addUserToFirestore = async (userData) => {
+  //   const addUser = doc(firestore, `users/${userData.email}`);
+  //   try {
+  //     await setDoc(addUser, userData, { merge: true });
+  //   } catch (error) {
+  //     alertDispatchHandler(alertDispatch, "ALERT", "INFO", error.message);
+  //   }
+  // };
+
```
Comment (yashodaAgarwal) [inline src/Context/Network/NetworkContext.jsx]: Remove the commented code before pushing code to production

File: src/Context/Network/NetworkContext.jsx [hunk 2]
```diff
@@ -30,6 +78,12 @@ const NetworkProvider = ({ children }) => {
           user: response.user.providerData[0],
         },
       });
```
Comment (yashodaAgarwal) [inline src/Context/Network/NetworkContext.jsx:51]: Great work to use Login by google

File: src/Components/UI/Modal/BadgeModal.css [hunk 3]
```diff
@@ -0,0 +1,21 @@
+.badge-modal-container {
+  position: fixed;
+  width: 100%;
+  height: 100vh;
+  left: 50%;
+  top: 50%;
+  transform: translate(-50%, -50%);
+  z-index: 4;
```
Comment (PuneetP16) [inline src/Components/UI/Modal/BadgeModal.css:8]: Suggestion: use variables instead of direct values.

File: src/Pages/Question/QuestionPage.jsx [hunk 4]
```diff
@@ -87,6 +131,59 @@ export const QuestionPage = () => {
...
+        } catch (error) {
+          alertDispatchHandler(alertDispatch, "ALERT", "INFO", error.message);
+        }
+      })();
+
+      // update winningStreak after celebration
+      // console.log(playedQuizData);
+      // updateFirestoreUserData(email, playedQuizData);
+
+      setTimeout(() => {
```
Comment (PuneetP16) [inline src/Pages/Question/QuestionPage.jsx]: remove the commented code.

**Review by PuneetP16 — APPROVED** [review]
Your overall app is looking great. it is a detailed app covering most of the production app functionality good work on this project.

File: src/Context/Network/NetworkContext.jsx (hunk 1 above)
Comment (saurabhchirde) [reply src/Context/Network/NetworkContext.jsx]: Noted

File: src/Pages/Question/QuestionPage.jsx (hunk 4 above)
Comment (saurabhchirde) [reply src/Pages/Question/QuestionPage.jsx]: Noted

**Pull request merged**